src/rp2_common/hardware_pio/pio.c

```diff
@@ -250,8 +268,17 @@ void pio_sm_set_pins_with_mask(PIO pio, uint sm, uint32_t pinvals, uint32_t pin_
 }
 
 void pio_sm_set_pindirs_with_mask(PIO pio, uint sm, uint32_t pindirs, uint32_t pin_mask) {
+    pio_sm_set_pindirs_with_mask64(pio, sm, pindirs, pin_mask);
+}
+
+void pio_sm_set_pindirs_with_mask64(PIO pio, uint sm, uint64_t pindirs, uint64_t pin_mask) {
     check_pio_param(pio);
     check_sm_param(sm);
+    check_pio_pin_mask64(pio, pin_mask);
```

Comment:
Should it also do `check_pio_pin_mask64(pio, pindirs);` ?

Reply:
no